impala/tests/test_data_types.py

```diff
@@ -54,43 +54,47 @@ def date_table(cur):
     table_name = 'tmp_date_table'
     ddl = """CREATE TABLE {0} (d date)""".format(table_name)
     cur.execute(ddl)
+    cur.execute('''insert into {0}
+                   values (date "0001-01-01"), (date "1999-9-9")'''.format(table_name))
     try:
         yield table_name
     finally:
         cur.execute("DROP TABLE {0}".format(table_name))
 
-
-@pytest.mark.connect
-def test_date_basic(cur, date_table):
+def setup_test_date(cur, date_table):
     """Insert and read back a couple of data values in a wide range."""
-    cur.execute('''insert into {0}
-                   values (date "0001-01-01"), (date "1999-9-9")'''.format(date_table))
     cur.execute('select d from {0} order by d'.format(date_table))
     results = cur.fetchall()
     assert results == [(datetime.date(1, 1, 1),), (datetime.date(1999, 9, 9),)]
 
+@pytest.mark.connect
+def test_date_basic(cur, date_table):
+    setup_test_date(cur, date_table)
+
+@pytest.mark.connect
+def test_date_no_string_conv(cur_no_string_conv, date_table):
```

Comment:
Can you also add similar test for decimals?

Reply:
Sure